src/backends/kubernetes_backend.py

```diff
@@ -0,0 +1,353 @@
+from .backend_interface import BackendInterface, ServiceType
+from pathlib import Path
+
+from kubernetes import client, config
+from kubernetes.client.models.v1_pod import V1Pod
+from kubernetes.stream import stream
+
+import time
+import re
+import yaml
+import io
+
+from typing import cast
+
+from warnet.tank import Tank
+from warnet.status import RunningStatus
+from warnet.utils import parse_raw_messages, default_bitcoin_conf_args
+
+DOCKER_REGISTRY_CORE = "bitcoindevproject/k8s-bitcoin-core"
+DOCKER_REGISTRY_LND = "lightninglabs/lnd:v0.17.0-beta"
+POD_PREFIX = "tank"
+BITCOIN_CONTAINER_NAME = "bitcoin"
+LN_CONTAINER_NAME = "ln"
+
+class KubernetesBackend(BackendInterface):
+
+    def __init__(self, config_dir: Path, network_name: str, namespace = "default", logs_pod = "fluentd") -> None:
+        super().__init__(config_dir)
+        # assumes the warnet rpc server is always
+        # running inside a k8s cluster as a statefulset
+        config.load_incluster_config()
+        self.client = client.CoreV1Api()
+        self.namespace = namespace
+        self.logs_pod = logs_pod
+        self.network_name = network_name
+
+    def build(self) -> bool:
+        # TODO: just return true for now, this is so we can be running either docker or k8s as a backend
+        # on the same branch
+        return True
+
+    def up(self, warnet) -> bool:
+        self.deploy_pods(warnet)
+        return True
+
+    def down(self, warnet) -> bool:
+        """
+        Bring an exsiting network down.
+            e.g. `k delete -f warnet-tanks.yaml`
+        """
+        for tank in warnet.tanks:
+            self.client.delete_namespaced_pod(tank.container_name, self.namespace)
```

Comment:
I don't think `tank.container_name` is defined any more. When I try `warcli network down` in k8s mode I get an error:

```
Exception Exception in function 'wrapper' with context (args=(<warnet.warnet.Warnet object at 0xffff7fa7e880>,), kwargs={}, e=This method isn't implemented yet, func_name=wrapper, func=<function 
Warnet.warnet_down at 0xffff7fd35670>): This method isn't implemented yet

```

Reply:
hm, this error is because `warnet_down` is not implemented in the K8s backend class, but is present in the abstract base class.. but you might also be right about `container_name`